definitions/TGSA.def

```diff
@@ -0,0 +1,115 @@
+    apt-key adv --keyserver keyserver.ubuntu.com --recv-keys F60F4B3D7FA2AF80
+    apt-key adv --keyserver keyserver.ubuntu.com --recv-keys A4B469963BF863CC
+    sudo apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/3bf863cc.pub
+    sudo apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64/7fa2af80.pub
+
+    #apt-key adv --keyserver keyserver.ubuntu.com --recv-keys F60F4B3D7FA2AF80
+    #apt-key adv --keyserver keyserver.ubuntu.com --recv-keys A4B469963BF863CC
+    #apt-get update
+    #apt-get install -y git
+    #apt-get install -y wget
+    #apt-get install -y gnupg
+    # install gpu fix and clean up
+    cd /
+    chmod +x Singularity_gpu_fix.sh
+    ./Singularity_gpu_fix.sh
+    rm Singularity_gpu_fix.sh
+
+    # create default internal candle_data_dir, map external candle_data_dir here
+
+    # install python modules and model prerequisites
+    pip install git+https://github.com/ECP-CANDLE/candle_lib@develop
+
+    #pip install CPython==3.8.0
+    pip install torch-geometric==1.7.1
+    pip install torch_cluster==1.5.9
+    pip install torch_scatter==2.0.9
+    pip install torch_sparse==0.6.12
+    pip install torch_spline_conv==1.2.1
+    #wget https://data.pyg.org/whl/torch-1.9.0%2Bcu102/torch_cluster-1.5.9-cp38-cp38-linux_x86_64.whl
+    #wget https://data.pyg.org/whl/torch-1.9.0%2Bcu102/torch_scatter-2.0.9-cp38-cp38-linux_x86_64.whl
+    #wget https://data.pyg.org/whl/torch-1.9.0%2Bcu102/torch_sparse-0.6.12-cp38-cp38-linux_x86_64.whl
+    #wget https://data.pyg.org/whl/torch-1.9.0%2Bcu102/torch_spline_conv-1.2.1-cp38-cp38-linux_x86_64.whl
+
+    wget https://data.pyg.org/whl/torch-1.12.0%2Bcu113/torch_cluster-1.6.0%2Bpt112cu113-cp38-cp38-linux_x86_64.whl
+    wget https://data.pyg.org/whl/torch-1.12.0%2Bcu113/torch_scatter-2.1.0%2Bpt112cu113-cp38-cp38-linux_x86_64.whl
+    wget https://data.pyg.org/whl/torch-1.12.0%2Bcu113/torch_sparse-0.6.15%2Bpt112cu113-cp38-cp38-linux_x86_64.whl
+    wget https://data.pyg.org/whl/torch-1.12.0%2Bcu113/torch_spline_conv-1.2.1%2Bpt112cu113-cp38-cp38-linux_x86_64.whl
+
+    #pip install *.whl
+
+    pip install rdkit
+    pip install fitlog
+    pip install torch==1.9.0+cu102 torchvision==0.10.0+cu102 torchaudio===0.9.0 -f https://download.pytorch.org/whl/torch_stable.html
+
+    pip install dgllife==0.3.2
+    pip install dgl==0.9.0
+    pip install numpy==1.21.5
+    pip install pandas==1.3.5
+    pip install scikit-learn==1.0.2
+    pip install scikit-image==0.16.2
+    pip install networkx==2.6.3
+    pip install h5py==3.8.0
+
+    #rm ./*.whl
+
+    #conda env create -f TGSA/singularity/TGSA.yml
+
+    # setup model and executables
+    cd /usr/local/
+    git clone https://github.com/smujiang/TGSA.git
+
+    cp TGSA/data/enterez_NCBI_to_hugo_gene_symbol_march_2019.txt /candle_data_dir/
```

Comment:
This will fail when mounting into /candle_data_dir, the config file won't be there.